staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/marshal.go

```diff
@@ -59,6 +60,60 @@ func (s *JSONSchemaPropsOrBool) UnmarshalJSON(data []byte) error {
 	return nil
 }
 
+type cborMajorType int
+
+const (
+	// https://www.rfc-editor.org/rfc/rfc8949.html#section-3.1
+	cborUnsignedInteger cborMajorType = 0
+	cborNegativeInteger               = 1
+	cborByteString                    = 2
+	cborTextString                    = 3
+	cborArray                         = 4
+	cborMap                           = 5
+	cborTag                           = 6
+	cborOther                         = 7
+)
+
+const (
+	cborFalseVaue = 0xf4
+	cborTrueValue = 0xf5
+	cborNullValue = 0xf6
+)
+
+func cborType(b byte) cborMajorType {
+	return cborMajorType(b >> 5)
+}
+
+func (s JSONSchemaPropsOrBool) MarshalCBOR() ([]byte, error) {
+	if s.Schema != nil {
+		return cbor.Marshal(s.Schema)
+	}
+	return cbor.Marshal(s.Allows)
+}
+
+func (s *JSONSchemaPropsOrBool) UnmarshalCBOR(data []byte) error {
+	switch {
+	case len(data) == 0, data[0] == cborNullValue:
+		return nil
```

Comment:
Same for JSONSchemaPropsOrBool's UnmarshalJSON, `[]byte("null")` appears to zero out all fields. The only UnmarshalJSON implementation I see that implements the conventional no-op for `null` is `JSON`.